diff --git a/node_modules/jsqr/dist/jsQR.js b/node_modules/jsqr/dist/jsQR.js
index 429896a..dac2090 100644
--- a/node_modules/jsqr/dist/jsQR.js
+++ b/node_modules/jsqr/dist/jsQR.js
@@ -343,6 +343,7 @@ function scan(matrix) {
                 data: decoded.text,
                 chunks: decoded.chunks,
                 version: decoded.version,
+                errorCorrectionLevel: decoded.errorCorrectionLevel,
                 location: {
                     topRightCorner: extracted.mappingFunction(location_1.dimension, 0),
                     topLeftCorner: extracted.mappingFunction(0, 0),
@@ -788,7 +789,11 @@ function decodeMatrix(matrix) {
         }
     }
     try {
-        return decodeData_1.decode(resultBytes, version.versionNumber);
+        var decodeResult = decodeData_1.decode(resultBytes, version.versionNumber);
+        // PATCH: Include error correction level in the result
+        var ecLevelMap = ['M', 'L', 'H', 'Q'];
+        decodeResult.errorCorrectionLevel = ecLevelMap[formatInfo.errorCorrectionLevel] || null;
+        return decodeResult;
     }
     catch (_a) {
         return null;
